### PR TITLE
[ROCM] Support xla.gpu.cuda.graph.launch

### PR DESCRIPTION
**Pull request by draganmladjenovic**
Remove the cuda infix where possible

### CODE REVIEW

File: xla/debug_options_flags.cc [hunk 1]
```diff
@@ -79,13 +79,13 @@ DebugOptions DefaultDebugOptionsIgnoringFlags() {
   // flag.
   opts.set_xla_gpu_enable_cublaslt(false);
 
-  // TODO(b/258036887): Enable cuda_graph_level=2. Currently blocked by CUDA 12
+  // TODO(b/258036887): Enable gpu_graph_level=2. Currently blocked by CUDA 12
   // integration.
-  opts.set_xla_gpu_cuda_graph_level(1);
-  opts.set_xla_gpu_cuda_graph_instantiation_threshold(2);
+  opts.set_xla_gpu_graph_level(1);
```
Comment (ezhulenev) [inline xla/debug_options_flags.cc:84]: Can you please sync with latest version, I've just reverted it back to 0.

File: xla/stream_executor/gpu/gpu_graph.cc [hunk 2]
```diff
@@ -0,0 +1,242 @@
...
+          << ++num_launches_;
+
+  RETURN_IF_GPU_GRAPH_ERROR(GpuGraphLaunch(get(), AsGpuStreamValue(stream)),
+                            "failed to run gpu graph");
+
+  return tsl::OkStatus();
+}
+
+//===----------------------------------------------------------------------===//
+// CUDA Graph Helpers.
```
Comment (anlunx) [inline xla/stream_executor/gpu/gpu_graph.cc:149]:
```suggestion
// GPU Graph Helpers.
```

Comment (anlunx) [inline xla/stream_executor/gpu/gpu_graph.cc:149]:
```suggestion
// GPU Graph Helpers.
```

File: xla/stream_executor/gpu/gpu_graph.cc [hunk 3]
```diff
@@ -0,0 +1,242 @@
...
+//===----------------------------------------------------------------------===//
+
+tsl::StatusOr<OwnedGpuGraph> CaptureGpuGraph(
+    stream_executor::Stream* stream, absl::AnyInvocable<tsl::Status()> capture) {
+  VLOG(3) << "Capture gpu graph on a stream: "
+          << stream->DebugStreamPointers();
+
+  GpuGraphHandle graph;
+
+  // Get the underlying stream for passing to CUDA APIs.
```
Comment (anlunx) [inline xla/stream_executor/gpu/gpu_graph.cc:159]:
```suggestion
  // Get the underlying stream for passing to GPU runtime APIs.
```

File: xla/xla.proto [hunk 4]
```diff
@@ -439,20 +439,20 @@ message DebugOptions {
   // 3+   Enable cuda graphs for collectives.
```
Comment (anlunx) [inline xla/xla.proto:439]: There are some comments that should be updated in several files in this patch

**Comment by ezhulenev** [comment]
XLA Linux GPU fails with ```[./xla/service/gpu/runtime/graph_launch.h:29](https://cs.corp.google.com/piper///depot/google3/xla/service/gpu/runtime/graph_launch.h?l=29):10: fatal error: 'xla/stream_executor/gpu/gpu_graph.h' file not found
#include "xla/stream_executor/gpu/gpu_graph.h"
         ^~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~```

I suspect some guards were not updated?

**Comment by ezhulenev** [comment]
And now it's 

```
[xla/service/gpu/runtime/kernel_launch.cc:33] fatal error: 'xla/stream_executor/gpu/gpu_graph.h' file not found
#include "xla/stream_executor/gpu/gpu_graph.h"
```

**Review by ezhulenev — APPROVED** [review]

**Comment by ezhulenev** [comment]
Can you please sync to latest version, and then I'll try to submit it internally after fixing remaining issues, I don't think I'll get an internal change until you'll resolve all conflicts

**Comment by i-chaochen** [comment]
Since this PR is approved and passed all CI 

@ddunl wonder this PR can be merged now?

**Comment by ddunl** [comment]
Thank you for pinging on this, it wasn't picked up internally for some reason. I'll let you know if there's any issues once we are running tests

**Comment by ddunl** [comment]
Working on the internal patch now, hoping to have it merged tomorrow. Thanks for the PR!

**Comment by ddunl** [comment]
Sorry this is taking extra time because I need to rebase internally. Should be in by end of the week

**Comment by ddunl** [comment]
Alright sorry about this, it seems like this part of the code base is very active at the moment. Could you rebase? I'll apply my internal patch tomorrow.

**Comment by ddunl** [comment]
Trying to submit the patched version now. Hopefully no more rebasing :-)

**Comment by ddunl** [comment]
Thanks for this PR!! I want to add that we've been accepting PRs with the `check_contents` failing, but we are going to try to do that less in the future since it's adding some extra work for internal cleanups. So on future PRs it would be great if you could try to get check_contents to pass. Thanks for this PR again!!!

**Pull request closed without merging**